### ISSUE TITLE
Assertion failure at (object)->IsUint8Array() on Node.js v4.2.0

### ISSUE THREAD

**Issue by mikemorris** [mapbox/mapbox-gl-native#2597]
I'm seeing an assertion failure at https://github.com/nodejs/node/blob/master/src/node_buffer.cc#L151 causing Travis Node.js v4.2.0 tests to fail (previously passing on v4.1.2). I think this is related to [changes to native buffer checks](https://github.com/nodejs/node/pull/3080) in Node.js v4.2.0.

```
Assertion failed: ((object)->IsUint8Array()), function WeakCallback, file ../src/node_buffer.cc, line 151.
Process 54799 stopped
* thread #1: tid = 0xdff217, 0x00007fff90e97286 libsystem_kernel.dylib`__pthread_kill + 10, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00007fff90e97286 libsystem_kernel.dylib`__pthread_kill + 10
libsystem_kernel.dylib`__pthread_kill:
->  0x7fff90e97286 <+10>: jae    0x7fff90e97290            ; <+20>
    0x7fff90e97288 <+12>: movq   %rax, %rdi
    0x7fff90e9728b <+15>: jmp    0x7fff90e92c53            ; cerror_nocancel
    0x7fff90e97290 <+20>: retq
(lldb) bt
* thread #1: tid = 0xdff217, 0x00007fff90e97286 libsystem_kernel.dylib`__pthread_kill + 10, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
  * frame #0: 0x00007fff90e97286 libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007fff9836b9f9 libsystem_pthread.dylib`pthread_kill + 90
    frame #2: 0x00007fff981059b3 libsystem_c.dylib`abort + 129
    frame #3: 0x00007fff980cda99 libsystem_c.dylib`__assert_rtn + 321
    frame #4: 0x000000010068ac7e node`node::Buffer::CallbackInfo::WeakCallback(v8::Isolate*, v8::Local<v8::Object>) + 238
    frame #5: 0x0000000100315b41 node`v8::internal::GlobalHandles::Node::PostGarbageCollectionProcessing(v8::internal::Isolate*) + 225
    frame #6: 0x0000000100314864 node`v8::internal::GlobalHandles::PostGarbageCollectionProcessing(v8::internal::GarbageCollector) + 148
    frame #7: 0x0000000100338a11 node`v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) + 1505
    frame #8: 0x000000010033813f node`v8::internal::Heap::CollectGarbage(v8::internal::GarbageCollector, char const*, char const*, v8::GCCallbackFlags) + 703
    frame #9: 0x00000001003377a1 node`v8::internal::Heap::HandleGCRequest() + 177
    frame #10: 0x00000001002ed298 node`v8::internal::StackGuard::HandleInterrupts() + 104
    frame #11: 0x0000000100549b65 node`v8::internal::Runtime_StackGuard(int, v8::internal::Object**, v8::internal::Isolate*) + 53
    frame #12: 0x00000f1461506295
    frame #13: 0x00000f1461826f4b
    frame #14: 0x00000f14616e75c0
    frame #15: 0x00000f146151981d
    frame #16: 0x00000f14616e72cb
    frame #17: 0x00000f1461519494
    frame #18: 0x00000f14616e538a
    frame #19: 0x00000f14616e4ce3
    frame #20: 0x00000f1461519494
    frame #21: 0x00000f14616e4b7e
    frame #22: 0x00000f14616e41e7
    frame #23: 0x00000f14616e378a
    frame #24: 0x00000f1461519f7d
    frame #25: 0x00000f14615189e2
    frame #26: 0x00000001002ebaac node`v8::internal::Invoke(bool, v8::internal::Handle<v8::internal::JSFunction>, v8::internal::Handle<v8::internal::Object>, int, v8::internal::Handle<v8::internal::Object>*) + 732
    frame #27: 0x00000001001665e4 node`v8::Function::Call(v8::Local<v8::Context>, v8::Local<v8::Value>, int, v8::Local<v8::Value>*) + 276
    frame #28: 0x000000010067c018 node`node::MakeCallback(node::Environment*, v8::Local<v8::Value>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*) + 637
    frame #29: 0x00000001006837c8 node`node::CheckImmediate(uv_check_s*) + 98
    frame #30: 0x00000001007aa6b3 node`uv__run_check + 33
    frame #31: 0x00000001007a6369 node`uv_run + 343
    frame #32: 0x00000001006830d0 node`node::Start(int, char**) + 573
    frame #33: 0x0000000100001234 node`start + 52
```

**Comment by mikemorris** [comment]
Bug in Node.js v4.2.0 release, should be fixed by https://github.com/nodejs/node/pull/3329, will close when this lands in a new Node.js release.

**Comment by mikemorris** [comment]
Fixed in Node.js v4.2.1 https://github.com/nodejs/node/pull/3337

**Comment by mikemorris** [comment]
Closed this prematurely, v4.2.1 hasn't been released yet.
